src/PostSharp.Engineering.Sdk/AssemblyMetadata.targets

```diff
@@ -8,34 +8,11 @@
             <_Parameter2>$(Version)</_Parameter2>
         </AssemblyAttribute>
 
-    </ItemGroup>
-
-    <Target Name="_SetAssemblyBuildDateMetadataTouchFileProperty" BeforeTargets="_SetAssemblyBuildDateMetadata">
-        <PropertyGroup>
-            <!-- This touch file ensures that the metadata attributes are not appended again when the generated assembly info file has not changed. -->
-            <_SetAssemblyBuildDateMetadataTouchFile>$(GeneratedAssemblyInfoFile).postsharp_metadata_touch</_SetAssemblyBuildDateMetadataTouchFile>
-        </PropertyGroup>
-    </Target>
-
-    <Target Name="_SetAssemblyBuildDateMetadata" BeforeTargets="CoreCompile" DependsOnTargets="GenerateAssemblyInfo" Inputs="$(GeneratedAssemblyInfoFile)" Outputs="$(_SetAssemblyBuildDateMetadataTouchFile)" Condition="'$(GenerateAssemblyInfo)'=='true'">
-        <Exec Command="git log -1 --format=%25%25cd --date=iso-strict" WorkingDirectory="$(MSBuildProjectDirectory)" ConsoleToMSBuild="true" EchoOff="true" StandardOutputImportance="low">
-            <Output TaskParameter="ConsoleOutput" PropertyName="_PackageBuildDate" />
-        </Exec>
-
-        <PropertyGroup>
-            <_PackageBuildDateAssemblyMetadata></_PackageBuildDateAssemblyMetadata>
-            <_PackageBuildDateAssemblyMetadata Condition="'$(Language)'=='C#'">[assembly: System.Reflection.AssemblyMetadataAttribute(&quot;PackageBuildDate&quot;, &quot;$(_PackageBuildDate)&quot;)] // Generated by PostSharp.Engineering SDK's AssemblyMetadata.targets</_PackageBuildDateAssemblyMetadata>
-            <_PackageBuildDateAssemblyMetadata Condition="'$(Language)'=='VB'">&lt;Assembly: System.Reflection.AssemblyMetadataAttribute(&quot;PackageBuildDate&quot;, &quot;$(_PackageBuildDate)&quot;)&gt; 'Generated by PostSharp.Engineering SDK's AssemblyMetadata.targets</_PackageBuildDateAssemblyMetadata>
-        </PropertyGroup>
-
-        <WriteLinesToFile
-            File="$(GeneratedAssemblyInfoFile)"
-            Lines="$(_PackageBuildDateAssemblyMetadata)"
-            Overwrite="false"
-            Encoding="UTF-8"
-            Condition="'$(_PackageBuildDateAssemblyMetadata)'!=''" />
+        <AssemblyAttribute Include="System.Reflection.AssemblyMetadataAttribute">
+            <_Parameter1>PackageBuildDate</_Parameter1>
+            <_Parameter2>$(BuildDate)</_Parameter2>
```

Comment:
It's not good that this relies on a user-defined property. This is not reliable. A better design is that the BuildDate property is set by the prepare step, so it does not depend on the user.